### ISSUE TITLE
Best way to register a specific model on pod startup ? Kubernetes

### ISSUE THREAD

**Issue by Leoduv** [pytorch/serve#528]
Hi,

### Context 
I am currently trying to create a Kubernetes infrastructure using torchserve for the inference of some models. I have a loadbalancer, a horizontal autoscaling and the deployment with the .mar files in a volume ; but having a load balancer creates an unexpected problem : you have to send a curl request to register a model X, but the load balancer will only send it to a single pod. Meaning that if I have 4 torchserve pods running, only 1 of them will get the instruction to deploy the model X. So the 3 others won't be able to handle the predictions. 

### Problem 
I am looking for a way to have the pods register model X when they start. I tried using something like "lifecycle: postStart: *send curl request to register model*" but it didn't work (curl not found, I'm using the official torchserve image). 

I just wanted to know if someone tackled the same problem or if you have other ideas/solutions on how to do this.

Ps : is there a Slack or something like that where we can talk with the devs ? Like there is for Kubernetes for example

**Comment by maaquib** [comment]
Related to https://github.com/pytorch/serve/issues/88
@Leoduv Please track the above issue for k8s deployement feature progress
You can join the torchserve channel on Pytorch Slack for questions to the devs https://pytorch.org/resources/

**Comment by chauhang** [comment]
@Leoduv Please check https://github.com/pytorch/serve/pull/551 for K8s solution.

**Comment by yuanshaochen** [comment]
I have a similar use case as @Leoduv. I have multiple torchserve pods running and want to add more models for all pods using curl management API.

By looking into K8s solution, it could resolve registering a model on all pods during startup, but it does not talk about how to updated models for all pods during torchserve running. If I understand correctly, to add/update a model after torchserve startup, we have to update the snapshot in `config.properties` and then trigger a new deployment instead of using management API?

**Comment by Wilson-inclaims** [comment]
Hi, @Leoduv. I have the same issue. In my case, after new pods are generated, those seem to start serving.
Among some of the pods, only one pods' CPU usage was almost 100%, but the pod is not constant.
It was changed randomly, so I've found that the new pods were ready to serve.
And I checked that there wasn't any something wrong with the load balancer setting with AWS EKS engineers.
Did you get success to deploy the trochserve to Kubernetes finally?
